pkg/services/vimmachine.go

```diff
@@ -351,9 +363,9 @@ func (v *VimMachineService) createOrPatchVSphereVM(vimMachineCtx *capvcontext.VI
 		Namespace: vm.Namespace,
 		Name:      vm.Name,
 	}
-	result, err := ctrlutil.CreateOrPatch(vimMachineCtx, vimMachineCtx.Client, vm, mutateFn)
+	result, err := ctrlutil.CreateOrPatch(ctx, v.Client, vm, mutateFn)
 	if err != nil {
-		vimMachineCtx.Logger.Error(
+		log.Error(
 			err,
 			"failed to CreateOrPatch VSphereVM",
 			"namespace",
```

Comment:
Let's add the kv pair with `VSphereVM` and klog ObjRef  after initializing the `vm` struct above and drop the `namespace` + `name` parts here.